flytekit/types/structured/structured_dataset.py

```diff
@@ -279,14 +299,17 @@ def to_literal(
         # StructuredDataset instance.
         if issubclass(python_type, StructuredDataset):
             assert isinstance(python_val, StructuredDataset)
-            df_type = type(python_val.dataframe)
+            python_val.uri = python_val.uri or ctx.file_access.get_random_remote_path()
+            protocol = protocol_prefix(python_val.uri)
             format = python_val.file_format
-            protocol = self.DEFAULT_PROTOCOLS[df_type]
-            try:
-                protocol = protocol_prefix(python_val.uri)
-            except ValueError:
-                ...
-
+            if protocol != StructuredDatasetProtocol.BIGQUERY:
+                format = python_val.file_format or StructuredDatasetFileFormat.PARQUET
+            if python_val.dataframe is None:
```

Comment:
this is still under the `if issubclass(python_type, StructuredDataset)` block right?  If that is true, and the python_val is an instance of `StructuredDataset` - shouldn't we just raise an exception?

Reply:
when people do something below, `StructuredDataset.dataframe` will be empty here. so I directly return Literal here instead of invoking the encoder or raising an exception.
```
@task
def t3(dataset: StructuredDataset[my_cols]) -> StructuredDataset[my_cols]:
    return dataset
```